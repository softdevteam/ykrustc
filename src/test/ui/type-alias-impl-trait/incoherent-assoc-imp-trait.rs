// Regression test for issue 67856

#![feature(unboxed_closures)]
// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete
#![feature(fn_traits)]

trait MyTrait {}
impl MyTrait for () {}

impl<F> FnOnce<()> for &F {
    //~^ ERROR conflicting implementations
    //~| ERROR type parameter `F` must be used
    type Output = impl MyTrait;
    extern "rust-call" fn call_once(self, _: ()) -> Self::Output {}
}
fn main() {}
