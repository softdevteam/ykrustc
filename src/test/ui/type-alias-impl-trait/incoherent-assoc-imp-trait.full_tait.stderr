warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/incoherent-assoc-imp-trait.rs:6:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error[E0119]: conflicting implementations of trait `std::ops::FnOnce<()>` for type `&_`
  --> $DIR/incoherent-assoc-imp-trait.rs:13:1
   |
LL | impl<F> FnOnce<()> for &F {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<A, F> FnOnce<A> for &F
             where F: Fn<A>, F: ?Sized;

error[E0210]: type parameter `F` must be used as the type parameter for some local type (e.g., `MyStruct<F>`)
  --> $DIR/incoherent-assoc-imp-trait.rs:13:6
   |
LL | impl<F> FnOnce<()> for &F {
   |      ^ type parameter `F` must be used as the type parameter for some local type
   |
   = note: implementing a foreign trait is only possible if at least one of the types for which it is implemented is local
   = note: only traits defined in the current crate can be implemented for a type parameter

error: aborting due to 2 previous errors; 1 warning emitted

Some errors have detailed explanations: E0119, E0210.
For more information about an error, try `rustc --explain E0119`.
